apps/common/models.py

```diff
@@ -58,3 +58,39 @@ class Meta:
 
     def __str__(self):
         return self.name
+
+
+class Notification(models.Model):
+    ''' Generic Notification Model '''
+
+    class NotificationType(models.TextChoices):
+        ORDER_RECEIVED = 'order_received', 'Order received'
+        ORDER_PACKED = 'order_packed', 'Order packed'
+        ORDER_COMPLETED = 'order_completed', 'Order completed'
+        ORDER_CANCELLED = 'order_cancelled', 'Order cancelled'
+        GENERAL = 'general', 'General'
+
+    notification_type = models.CharField(
+        choices=NotificationType.choices, max_length=40,
+        default=NotificationType.GENERAL,
+        verbose_name=_("Notification Type")
+    )
+    title = models.TextField(verbose_name=_('Title'))
+    # Display to user
+    for_user = models.ForeignKey('user.User', on_delete=models.CASCADE, verbose_name=_('For user'),)
+    # Unread/Read
+    read = models.BooleanField(
+        default=False, verbose_name=_('Read'), help_text=_('Whether notification has been marked as read')
+    )
+    created_at = models.DateTimeField(
+        auto_now_add=True, verbose_name=_('Created at'), help_text=_('When notification was created')
+    )
+    # To describe notification
+    meta = models.JSONField(default=None, blank=True, null=True)
```

Comment:
We need to discuss this further, storing in JSON can cause issues if we need to attach foreign when using notification.